### PR TITLE
fix(to_pandas): handle empty datachain in to_pandas (and show)

### PR DESCRIPTION
**Pull request by shcheklein**
Closes https://github.com/iterative/datachain/issues/237 + simplifies the code a bit and makes sure that column names are preserved in `df` even if there are no values - consistently across flatten and non-flatten version.

```
files = [File(size=s, path=p) for p, s in zip(list("abcde"), range(5))]
DataChain.from_values(file=files).limit(0).show()
```

returns:

```
Empty result
Columns: [('file', 'source'), ('file', 'path'), ('file', 'size'), ('file', 'version'), ('file', 'etag'), ('file', 'is_latest'), ('file', 'last_modified'), ('file', 'location'), ('file', 'vtype')]
```

(similar to how pandas returns the list of columns to make it easier to debug)

Once thing that I was not able to imitate for now is a schema-less (no columns, no-values DataChain), but let's consider this as a minor edge case that we can test / fix later if we decide to allow that.

### CODE REVIEW

File: tests/func/test_dataset_query.py [hunk 1]
```diff
@@ -1185,8 +1185,8 @@ def name_len_interrupt(_name):
 
     q = (
         DatasetQuery(name="animals", version=1, catalog=catalog)
-        .filter(C.size < 13)
-        .filter(C.parent.glob("cats*") | (C.size < 4))
+        # .filter(C.size < 13)
```
Comment (shcheklein) [inline tests/func/test_dataset_query.py:1188]: disregard this, unrelated, this PR is primarily for @mattseddon and can be closed

Comment (mattseddon) [reply tests/func/test_dataset_query.py:1188]: closed my PR, you can take it

Comment (shcheklein) [reply tests/func/test_dataset_query.py:1188]: sounds good

**Review by mattseddon — APPROVED** [review]
